tasks/module_4/task_6.py

```diff
@@ -0,0 +1,25 @@
+"""Module4/Task6: Very sophisticated password breaker."""
+from PyPDF2 import PdfFileReader
+
+
+# Vars
+DICTIONARY_PATH = 'resources/task_6/dictionary.txt'
+PDF_PATH = 'resources/task_6/encrypted.pdf'
+
+
+def decrypt_pdf_password():
+    """Decrypt password for encrypted pdf from file of passwords."""
+    password = None
+
+    with open(DICTIONARY_PATH, 'r') as eng_dict, open(PDF_PATH, 'rb') as enc_pdf:
+        my_pdf = PdfFileReader(enc_pdf)
+        for word in eng_dict.read().splitlines():
+            if my_pdf.decrypt(word.upper()) or my_pdf.decrypt(word.lower()):
```

Comment:
Two points here:
- If you find password you should exit the loop. If first word from dictionary is the first password, you will loose some time checking the rest of the dictionary.
- Because you are checking both upper and lower word in one condition, your program cannot tell which one of them is correct password.